src/iexamine.cpp

```diff
@@ -3483,29 +3531,30 @@ void iexamine::autodoc( player &p, const tripoint &examp )
     };
 
     bool adjacent_couch = false;
-    bool in_position = false;
-    for( const auto &couch_loc : g->m.points_in_radius( examp, 1, 0 ) ) {
-        const furn_str_id couch( "f_autodoc_couch" );
-        if( g->m.furn( couch_loc ) == couch ) {
-            adjacent_couch = true;
-            if( p.pos() == couch_loc ) {
-                in_position = true;
-            }
-        }
-    }
+    static player null_player;
+    player &patient = player_on_couch( p, examp, null_player, adjacent_couch );
+    player &installer = best_installer( p );
 
     if( !adjacent_couch ) {
         popup( _( "No connected couches found.  Operation impossible.  Exiting." ) );
         return;
     }
-    if( !in_position ) {
+    if( &patient == &null_player ) {
         popup( _( "No patient found located on the connected couches.  Operation impossible.  Exiting." ) );
         return;
     }
 
-    const bool has_anesthesia = p.crafting_inventory().has_item_with( []( const item &it ) {
-        return it.has_flag( "ANESTHESIA" );
-    } );
+    const item_location anesthesia_kit = g->inv_map_splice( []( const item &e ) {
+                return e.has_flag( "ANESTHESIA" );
+    }, _( "Choose anesthesia kit" ), PICKUP_RANGE,
```

Comment:
Proposing to choose an anesthesia kit is an added keypress. Is there a real need for this not to be automatic?

Reply:
will fix as soon as I figure out how.

Reply:
It is automatic before. First the game checked for kit available, and then the game consumed it automatically through the `consume_items` function.

Reply:
Right, but player:: introduce_to_anesthesia is now called by the patient, who may be an NPC without anesthesia in his inventory. So I had to move the code to push the comp up to here.

Reply:
Then why don't check for anesthesia kit in NPC inventory?

Reply:
Because I think the use flow of "push the NPC onto the couch, activate autodoc, use CBM and kit from your inventory" is a lot more intuitive than "push the NPC onto the couch, give the NPC a CBM and anesthesia kit, activate autodoc, use CBM and kit from the NPC's inventory."